src/vcUnitConversion.cpp

```diff
@@ -61,3 +62,137 @@ double vcUnitConversion_ConvertTemperature(double sourceValue, vcTemperatureUnit
+    425865622.0,  // 1983-06-01
+    489024023.0,  // 1985-06-01
+    567993624.0,  // 1987-12-01
+    631152025.0,  // 1988-12-01
+    662688026.0,  // 1990-12-01
+    709948827.0,  // 1992-06-01
+    741484828.0,  // 1993-06-01
+    773020829.0,  // 1994-06-01
+    820454430.0,  // 1995-12-01
+    867715231.0,  // 1997-06-01
+    915148832.0,  // 1998-12-01
+    1136073633.0, // 2005-12-01
+    1230768034.0, // 2008-12-01
+    1341100835.0, // 2012-06-01
+    1435708836.0, // 2015-06-01
+    1483228837.0  // 2016-12-01
+  };
+
+  //Convert sourceValue to TAI
+  switch (sourceReference)
+  {
+    case vcTimeReference_TAI:
+    {
+      result.seconds = sourceValue.seconds;
+      break;
+    }
+    case vcTimeReference_Unix:
+    {
+      if (sourceValue.seconds < 0.0)
+        goto epilogue;
+
+      double leapSeconds = 0.0;
+      for (size_t i = 0; i < udLengthOf(s_leapSeconds); ++i)
+      {
+        if (sourceValue.seconds < s_leapSeconds[i])
+          break;
+        leapSeconds++;
+      }
+      result.seconds = s_secondsBetweenEpochs_TAI_Unix + sourceValue.seconds + leapSeconds;
+      break;
+    }
+    case vcTimeReference_GPS:
+    {
+      result.seconds = sourceValue.seconds + s_secondsBetweenEpochs_TAI_GPS;
+      break;
+    }
+    case vcTimeReference_GPSAdjusted:
+    {
+      result.seconds = sourceValue.seconds + (s_secondsBetweenEpochs_TAI_GPS + 1.0e9);
+      break;
+    }
+    case vcTimeReference_GPSWeek:
+    {
+      if (sourceValue.GPSWeek.secondsOfTheWeek < 0.0)
+        goto epilogue;
+
+      result.seconds = s_secondsBetweenEpochs_TAI_GPS + sourceValue.GPSWeek.secondsOfTheWeek + s_weekSeconds * double(sourceValue.GPSWeek.weeks);
+      break;
+    }
+  }
+
+  //Required Reference
+  switch (requiredReference)
+  {
+    case vcTimeReference_TAI:
+    {
+      break;
+    }
+    case vcTimeReference_Unix:
+    {
+      if (result.seconds < s_secondsBetweenEpochs_TAI_Unix)
+        goto epilogue;
+
+      result.seconds -= s_secondsBetweenEpochs_TAI_Unix;
+      double leapSeconds = 0.0;
+      for (size_t i = 0; i < udLengthOf(s_leapSeconds); ++i)
+      {
+        if (result.seconds < s_leapSeconds[i])
+          break;
+        leapSeconds++;
+      }
+      result.seconds = result.seconds - leapSeconds;
+      break;
+    }
+    case vcTimeReference_GPS:
+    {
+      result.seconds -= s_secondsBetweenEpochs_TAI_GPS;
+      break;
+    }
+    case vcTimeReference_GPSAdjusted:
+    {
+      result.seconds -= (s_secondsBetweenEpochs_TAI_GPS + 1.0e9);
+      break;
+    }
+    case vcTimeReference_GPSWeek:
+    {
+      if (result.seconds < s_secondsBetweenEpochs_TAI_GPS)
+        goto epilogue;
+
+      result.seconds -= s_secondsBetweenEpochs_TAI_GPS;
+      result.GPSWeek.weeks = (uint32_t)udFloor(result.seconds / s_weekSeconds);
+      result.GPSWeek.secondsOfTheWeek = result.seconds - (s_weekSeconds * result.GPSWeek.weeks);
+      break;
+    }
+  }
+
+  result.success = true;
+  epilogue:
```

Comment:
We usually format the epilogue like this:
```c++
  result.success = true;

epilogue:
  return result;
```

It's one indentation less (or it's no indentation at all, I'm not sure which since we only use labels at the end of a function) and it has a space before it to signify the separation between the "success" and "cleanup/end" code. The lack of space after is used to avoid non-cleanup/end code being put between them too.